containers/src/Kokkos_DynRankView.hpp

```diff
@@ -2063,10 +2073,21 @@ inline void realloc(DynRankView<T, P...>& v,
   static_assert(Kokkos::ViewTraits<T, P...>::is_managed,
                 "Can only realloc managed views");
 
-  const std::string label = v.label();
+  const size_t new_extents[8] = {n0, n1, n2, n3, n4, n5, n6, n7};
+  bool sizeMismatch           = false;
+  for (unsigned int dim = 0; dim < v.rank(); ++dim)
+    if (new_extents[dim] != v.extent(dim)) {
+      sizeMismatch = true;
+      break;
+    }
+
+  if (sizeMismatch) {
+    const std::string label = v.label();
 
-  v = drview_type();  // Deallocate first, if the only view to allocation
-  v = drview_type(label, n0, n1, n2, n3, n4, n5, n6, n7);
+    v = drview_type();  // Deallocate first, if the only view to allocation
+    v = drview_type(label, n0, n1, n2, n3, n4, n5, n6, n7);
+  } else
+    Kokkos::deep_copy(v, typename drview_type::value_type{});
```

Comment:
Same question about the else-branch.

Reply:
Citing from https://github.com/kokkos/kokkos/pull/4207#issue-701805574:
> For resize we don't do anything if the extents already match but we call Kokkos::deep_copy for realloc to preserve the current behavior. This should of course change when we add the AllocateWithoutInitializing overloads.

I expect `DynRankView` to also zero initialize in the constructor and hence expect that `realloc` was also zero initializing before.